openwisp_controller/config/api/filters.py

```diff
@@ -98,29 +94,32 @@ class DeviceListFilter(BaseConfigAPIFilter):
         lookup_expr='lt',
     )
 
-    def filter_devicelocation(self, queryset, name, value):
-        # Returns list of device that have devicelocation objects
-        return queryset.exclude(devicelocation__isnull=value)
-
     def _set_valid_filterform_lables(self):
         self.filters['group'].label = _('Device group')
         self.filters['config__templates'].label = _('Config template')
         self.filters['config__status'].label = _('Config status')
         self.filters['config__backend'].label = _('Config backend')
-        self.filters['with_geo'].label = _('Has geographic location set?')
 
     def __init__(self, *args, **kwargs):
         super(DeviceListFilter, self).__init__(*args, **kwargs)
         self._set_valid_filterform_lables()
 
+    if apps.is_installed('openwisp_controller.geo'):
+        from openwisp_controller.geo.api.filters import DeviceLocationFilter
+
+        def filter_devicelocation(self, queryset, name, value):
+            # Returns list of device that have devicelocation objects
+            return queryset.exclude(devicelocation__isnull=value)
+
+        with_geo = DeviceLocationFilter().filters['with_geo']
```

Comment:
The app `controller.config` must not know anything about `controller.geo`.
I will push an alternative implementation that is acceptable and consistent within OpenWISP which I want you to look at and learn from.